packages/playground/src/components/manage_gateway_dialog.vue

```diff
@@ -0,0 +1,280 @@
+        <v-card-text class="d-flex">
+          <v-chip color="error" label variant="tonal" class="mr-1 mb-1" v-for="gw in gatewaysToDelete" :key="gw.name">
+            {{ gw.name }}
+          </v-chip>
+        </v-card-text>
+
+        <v-divider />
+
+        <v-card-actions class="d-flex justify-end">
+          <v-btn
+            color="error"
+            variant="outlined"
+            size="small"
+            :disabled="loadingGateways || deleting"
+            @click="
+              () => {
+                requestDelete = false;
+                deleteSelectedGateways();
+              }
+            "
+            >Delete</v-btn
+          >
+          <v-btn color="secondary" variant="tonal" size="small" @click="requestDelete = false">Cancel</v-btn>
+        </v-card-actions>
+      </v-card>
+    </v-dialog>
+  </div>
+</template>
+
+<script lang="ts">
+import { onMounted, PropType, ref } from "vue";
+
+import { useProfileManager } from "../stores";
+import type { GatewayNode } from "../types";
+import { deployGatewayName, type GridGateway, loadDeploymentGateways } from "../utils/gateway";
+import { getGrid } from "../utils/grid";
+import { normalizeError } from "../utils/helpers";
+import { generateName } from "../utils/strings";
+import ListTable from "./list_table.vue";
+import SelectGatewayNode from "./select_gateway_node.vue";
+import { useLayout } from "./weblet_layout.vue";
+
+export interface VM {
+  deploymentName: string;
+  projectName: string;
+}
+
+export default {
+  name: "ManageGatewayDialog",
+  components: { SelectGatewayNode, ListTable },
+  props: {
+    vm: Object as PropType<VM>,
+  },
+  setup(props) {
+    const profileManager = useProfileManager();
+    const layout = useLayout();
+    const gatewayTab = ref(0);
+
+    const subdomain = ref<string>(generateName({}, 12));
+    const gatewayNode = ref<GatewayNode>();
+    const port = ref(80);
+    const useCustomDomain = ref(false);
+    const domain = ref<string>();
+    const valid = ref(false);
+
+    const ip = props.vm[0].interfaces[0].ip as string;
+    const networkName = props.vm[0].interfaces[0].network as string;
+
+    onMounted(loadGateways);
+
+    const loadingGateways = ref(false);
+    const gateways = ref<GridGateway[]>([]);
+    async function loadGateways() {
+      gateways.value = [];
+      gatewaysToDelete.value = [];
+      loadingGateways.value = true;
+      const grid = await getGrid(profileManager.profile!, props.vm.projectName);
+      gateways.value = await loadDeploymentGateways(grid);
+      loadingGateways.value = false;
+    }
+
+    async function deployGateway() {
+      layout.value.setStatus("deploy");
+
+      try {
+        const [x, y] = ip.split(".");
+        const grid = await getGrid(profileManager.profile!, props.vm.projectName);
+        await grid.networks
+          .addNode({ name: networkName, ipRange: `${x}.${y}.0.0/16`, nodeId: gatewayNode.value.id })
+          .catch(() => null);
```

Comment:
In case of failure while adding the node to the network, show the error message.